From eb0d549335c3490fae86b0d5fac932cf643456ac Mon Sep 17 00:00:00 2001
From: "Guoniu.zhou" <guoniu.zhou@nxp.com>
Date: Mon, 30 May 2022 15:09:26 +0800
Subject: [PATCH 116/274] LF-6257-05: arm64: dts: add regulators for AP1302 ISP
 daughter card

Add regulators for AP1302 ISP daughter card. They will be used to
control ISP daughter card power on sequnece.

Signed-off-by: Guoniu.zhou <guoniu.zhou@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
---
 .../boot/dts/freescale/imx93-11x11-evk.dts    | 56 +++++++++++++++++++
 1 file changed, 56 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts
index cb1b115e3..e6aa11d2a 100644
--- a/arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts
@@ -47,6 +47,62 @@ usdhc3_pwrseq: usdhc3_pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		reset-gpios = <&pcal6524 20 GPIO_ACTIVE_LOW>;
 	};
+
+	reg_dvdd_sel: regulator-dvdd_sel {
+		compatible = "regulator-fixed";
+		regulator-name = "DVDD_SEL";
+		gpio = <&adp5585gpio_isp 0 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		startup-delay-us = <2000>;
+	};
+
+	reg_dvdd_1v2: regulator-dvdd {
+		compatible = "regulator-fixed";
+		regulator-name = "DVDD_1V2";
+		gpio = <&adp5585gpio_isp 6 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <1200000>;
+		regulator-max-microvolt = <1200000>;
+		enable-active-high;
+		vin-supply = <&reg_dvdd_sel>;
+	};
+
+	reg_vdd_3v3: regulator-vdd {
+		compatible = "regulator-fixed";
+		regulator-name = "VDD_3V3";
+		gpio = <&adp5585gpio_isp 5 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		startup-delay-us = <4000>;
+		enable-active-high;
+	};
+
+	reg_vddio_1v8: regulator-vddo {
+		compatible = "regulator-fixed";
+		regulator-name = "VDDIO_1V8";
+		gpio = <&adp5585gpio_isp 9 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		startup-delay-us = <4000>;
+		enable-active-high;
+		vin-supply = <&reg_vdd_3v3>;
+	};
+
+	reg_vaa_sel: regulator-vaa_sel {
+		compatible = "regulator-fixed";
+		regulator-name = "VAA_SEL";
+		gpio = <&adp5585gpio_isp 1 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	reg_avdd_2v8: regulator-avdd {
+		compatible = "regulator-fixed";
+		regulator-name = "AVDD_2V8";
+		gpio = <&adp5585gpio_isp 7 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		enable-active-high;
+		vin-supply = <&reg_vaa_sel>;
+	};
 };
 
 &adc1 {
-- 
2.25.1

